azurelinuxagent/ga/exthandlers.py

```diff
@@ -2283,6 +2289,28 @@ def get_resource_limits(self, extension_name, str_version):
 
         return ResourceLimits(self.data.get('resourceLimits', None))
 
+    def report_invalid_boolean_properties(self, ext_name):
+        """
+        Check that the specified keys in the handler manifest has boolean values.
+        """
+        for key in ['reportHeartbeat', 'continueOnUpdateFailure', 'supportsMultipleExtensions']:
+            value = self.data['handlerManifest'].get(key, False)
+            if not isinstance(value, bool):
+                msg =("In the handler manifest: '{0}' has a non-boolean value [{1}] for boolean type. Please change it to a boolean value. "
+                      "For backward compatibility, 'true' (case insensitive) is accepted, and other values default to False.").format(key, value)
+                logger.warn(msg)
+                add_event(name=ext_name, is_success=False, message=msg, op=WALAEventOperation.ExtensionHandlerManifest, log_event=False)
+
+    @staticmethod
+    def _parse_boolean_value(value, default_val=False):
```

Comment:
let's not have a default for default_val; in this case, forcing the caller to explicitly specify the default value makes the code more readable